src/mscorlib/src/System/MulticastDelegate.cs

```diff
@@ -103,11 +104,22 @@ public override void GetObjectData(SerializationInfo info, StreamingContext cont
         [System.Security.SecuritySafeCritical]  // auto-generated
         public override sealed bool Equals(Object obj)
         {
-            if (obj == null || !InternalEqualTypes(this, obj))
-                return false;      
-            MulticastDelegate d = obj as MulticastDelegate;
-            if (d == null)
+            if (obj == null)
                 return false;
+            if (object.ReferenceEquals(this, obj))
+                return true;
+            
+            // TODO: See if this can be changed to GetType() == obj.GetType()
+            // without negatively impacting performance. GetType is a JIT
```

Comment:
The GetType JIT intrinsic is only optimized in specific usage patterns. This is not one of them.